elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -0,0 +1,244 @@
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.yahoo.elide.graphql.ModelBuilder.ARGUMENT_DATA;
+import static com.yahoo.elide.graphql.ModelBuilder.ARGUMENT_OPERATION;
+
+/**
+ * Interacts with {@link PersistentResource} to fetch data for GraphQL.
+ */
+@Slf4j
+public class PersistentResourceFetcher implements DataFetcher {
+    private final ElideSettings settings;
+
+    public PersistentResourceFetcher(ElideSettings settings) {
+        this.settings = settings;
+    }
+
+    @Override
+    public Object get(DataFetchingEnvironment environment) {
+        Map<String, Object> args = environment.getArguments();
+        Environment context = new Environment(environment);
+        RelationshipOp operation = (RelationshipOp) args.getOrDefault(ARGUMENT_OPERATION, RelationshipOp.FETCH);
+
+        if (log.isDebugEnabled()) {
+            logContext(operation, context);
+        }
+
+        switch (operation) {
+            case FETCH:
+                return fetchObject(context);
+
+            case ADD:
+                return createObject(context);
+
+            case DELETE:
+                return deleteObject(context);
+
+            case REPLACE:
+                return new UpdateOperation(context).execute();
+        }
+
+        throw new UnsupportedOperationException("Unknown operation: " + operation);
+    }
+
+    private void logContext(RelationshipOp operation, Environment environment) {
+        List<Field> children = environment.field.getSelectionSet() != null
+                ? (List) environment.field.getSelectionSet().getChildren()
+                : new ArrayList<>();
+        String requestedFields = environment.field.getName() + (children.size() > 0
+                ? "(" + children.stream().map(Field::getName).collect(Collectors.toList()) + ")" : "");
+        GraphQLType parent = environment.parentType;
+        log.debug("{} {} fields for {} with parent {}<{}>",
+                operation, requestedFields, environment.source, parent.getClass().getSimpleName(), parent.getName());
+    }
+
+    private Object deleteObject(Environment request) {
+        Set<Object> deleted = new HashSet<>();
+
+        if (!request.id.isPresent() && request.data.size() > 1) {
+            throw new WebApplicationException("Id argument specification with an additional list of id's to delete "
+                    + "is unsupported", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        EntityDictionary dictionary = request.requestScope.getDictionary();
+
+        String loadType = request.field.getName();
+        // TODO: This works at the root-level, but will it blow up in nested deletes?
+        String idFieldName = dictionary.getIdFieldName(dictionary.getEntityClass(loadType));
+        String loadId = request.field.getArguments().stream()
```

Comment:
I think we need to move some of this logic into the environment.  There should be common methods for getting the set of IDs that need to be loaded, etc.